`std.Target`: Rewrite DynamicLinker.standard() and fill in some missing details.

alexrp opened the pull request:
I went and collected all the known dynamic linker paths from GCC, Clang, and a few other places online, and then rewrote this function. Also moved some OS prongs for which I concluded with a reasonable degree of confidence that the OS doesn't have a standard dynamic linker path.

Recommend reviewing this one by just looking at the new implementation rather than the diff.

(Almost all of the ABI-related TODOs I'm adding here can be addressed by #20690.)

linusg commented on lib/std/Target.zig:
```diff
@@ -1706,155 +1706,297 @@ pub const DynamicLinker = struct {
+                    .x86 => init("/lib/ld-linux.so.2"),
+                    .x86_64 => init(if (abi == .gnux32) "/libx32/ld-linux-x32.so.2" else "/lib64/ld-linux-x86-64.so.2"),
+
+                    .xtensa => init("lib/ld.so.1"),
```
Looks like this is missing the leading slash?

alexrp replied:
Yep, fixed.

alexrp commented:
Unrelated hang on `aarch64-macos-debug`.

alexrp commented:
Linux failures are unrelated; would be good if someone can retry them.

linusg reviewed (approved):
LGTM, good TODO comments :+1:

alexrp commented:
No idea what's up with the Linux CI; doesn't seem related.

alexrp commented:
Speaking of TODOs, @andrewrk would you prefer that I remove them and just list out the missing bits in an issue, or is it fine to keep them on the understanding that, once implemented, #20690 will address them?

alexrp commented:
* Rebased and resolved conflicts.
* Addressed a few TODOs.
* Removed some TODOs that will fall out automatically when we update to LLVM 20.
* Remaining TODOs will either be addressed by #20690, or in some cases by a follow-up PR that adds a few more `Cpu.Arch` tags.

alexrp commented:
> or in some cases by a follow-up PR that adds a few more `Cpu.Arch` tags

Will address these after merging.
